File: in_Java/0447 Number of Boomerangs.java
class Solution {
    public int numberOfBoomerangs(int[][] points) {
        int res = 0;
        if(points.length < 3) return res;
        Map<Integer,Integer> hm;
        for(int i = 0; i < points.length; i++){
            hm = new HashMap<>();
            int a = points[i][0], b = points[i][1];
            for(int j = 0; j < points.length; j++){
                if(i == j) continue;
                int aa = points[j][0], bb = points[j][1];
                int dis = (aa-a) * (aa-a) + (bb-b) * (bb-b);
                hm.put(dis, hm.getOrDefault(dis,0)+1);
            }
            for(int v : hm.values()){
                if(v < 2) continue;
                res += v * (v-1) / 2;
            }
        }
        return res*2;
    }
}

File: in_Python/0447 Number of Boomerangs.py
class Solution(object):
    def numberOfBoomerangs(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        N = len(points)
        mat = []
        for i in range(N):
            c = []
            for j in range(N):
                c.append(0)
            mat.append(c)
        for i in range(N):
            cur = points[i]
            for j in range(i+1, N):
                node = points[j]
                d = (cur[0] - node[0]) * (cur[0] - node[0]) + (cur[1] - node[1]) * (cur[1] - node[1])
                mat[i][j] = d
                mat[j][i] = d
        res = 0
        for i in range(N):
            row = mat[i]
            d = {}
            for r in row:
                if r == 0:
                    continue
                if r not in d:
                    d[r] = 1
                else:
                    d[r] += 1
            for k in d:
                u = d[k]
                res += u * (u-1)
        return res
                    